src/app/shared/resizable-bottom-sheet/resizable-bottom-sheet.component.ts

```diff
@@ -0,0 +1,155 @@
+  @Output() openChange = new EventEmitter<boolean>();
+
+  closeThreshold = 0.5; // % of height reduction that closes overlay
+
+  animationState = 'closed';
+  resizingHeight = this.height;
+
+  constructor(private overlay: Overlay, private viewContainerRef: ViewContainerRef, private renderer: Renderer2) {}
+
+  @ViewChild('bottomSheet') bottomSheetRef;
+  @ViewChild('resizeHandle') resizeHandleRef: ElementRef;
+  @ViewChild('bottomSheetContainer') bottomSheetContainerRef: ElementRef;
+  overlayRef: OverlayRef = null;
+  portal: TemplatePortal = null;
+
+  private destroy = new Subject<void>();
+
+  ngAfterViewInit(): void {
+    this.portal = new TemplatePortal(this.bottomSheetRef, this.viewContainerRef);
+
+    if (!this.overlayRef) {
+      this.overlayRef = this.overlay.create({
+        hasBackdrop: true,
+        panelClass: 'resizable-bottom-sheet-overlay-panel',
+        backdropClass: 'resizable-bottom-sheet-overlay-backdrop',
+        disposeOnNavigation: true,
+        positionStrategy: this.overlay.position().global().centerHorizontally().bottom(),
+        scrollStrategy: this.overlay.scrollStrategies.block()
+      });
+
+      this.overlayRef
+        .backdropClick()
+        .pipe(takeUntil(this.destroy))
+        .subscribe(() => {
+          this.close();
+        });
+    }
+
+    if (this.open) {
+      setTimeout(() => {
+        this.overlayRef.attach(this.portal);
+        this.animationState = 'open';
+      }, 0);
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (!this.overlayRef) {
+      return;
+    }
+
+    if (!changes.open?.currentValue) {
+      this.overlayRef.detach();
+      return;
+    }
+
+    if (!this.overlayRef.hasAttached()) {
+      this.overlayRef.attach(this.portal);
+    }
+    this.animationState = 'open';
+  }
+
+  ngOnDestroy(): void {
+    this.destroy.next();
+    this.destroy.complete();
+  }
+
+  onHandleTouchStart(touchStartEvent: TouchEvent): void {
+    const startY = touchStartEvent.touches[0].clientY;
+    let delta = 0;
+    const startHeight = this.bottomSheetContainerRef.nativeElement.getBoundingClientRect().height;
+    this.animationState = 'resize';
+    let cleanupTouchMove = () => {};
+    let cleanupTouchEnd = () => {};
+    this.resizingHeight = startHeight;
+    cleanupTouchMove = this.renderer.listen(this.resizeHandleRef.nativeElement, 'touchmove', (touchMoveEvent) => {
+      const movedTo = touchMoveEvent.touches[0].clientY;
+      delta = movedTo - startY;
+      this.resizingHeight = startHeight - delta;
+      this.renderer.setStyle(this.bottomSheetContainerRef.nativeElement, 'height', `${startHeight - delta}px`);
+    });
+    cleanupTouchEnd = this.renderer.listen(this.resizeHandleRef.nativeElement, 'touchend', (touchEndEvent) => {
+      if (delta > startHeight * this.closeThreshold) {
+        this.close();
+      } else {
+        this.animationState = 'open';
+      }
+      cleanupTouchMove();
+      cleanupTouchEnd();
```

Comment:
What's the point of running local functions that just return an empty object?

Reply:
Functions are overridden to clean up subscriptions to `touchmove` and `touchend` events.